Makefile.dir-variables

```diff
@@ -0,0 +1,67 @@
+### This Makefile creates variables for all Contiki-NG directory locations
+### Instead of storing them in the top-level Makefile.include, we store them
+### in this separate Makefile here, which is included by Makefile.include
+### This allows examples that wish to specify additional MODULES to include
+### this Makefile and append to MODULES using the path variables defined here
+
+# Variables for the various directories under os/
+# Those are sometimes used relative to the top-level dir in which case we don't
+# need to prepend $(CONTIKI), as well as relative to the example dir, in which
+# case we do. So we define the variables to suit the former case and we will
+# prepend $(CONTIKI) explicitly where needed.
+CONTIKI_NG_OS_DIR = os
```

Comment:
shouldn't all the variables in this file be defined with `?=`? (e.g. for example makefile to override)

Reply:
So with the exception of `ARCH_PATH` we are not allowing the user to override the location of any other parts of the tree. Why would we want to start doing so now?